integration-tests/upstream/build.gradle

```diff
@@ -43,7 +43,17 @@ test.filter {
   excludeTest 'dagger.functional.BasicTest', null
 
   // TODO reflect bug! Generics don't work well.
-  excludeTest 'dagger.functional.GenericTest', null
+  excludeTest 'dagger.functional.GenericTest', 'boundedGenerics'
+  excludeTest 'dagger.functional.GenericTest', 'complexGenerics'
+  excludeTest 'dagger.functional.GenericTest', 'genericModules'
+  excludeTest 'dagger.functional.GenericTest', 'membersInjections'
+  excludeTest 'dagger.functional.GenericTest', 'noDepsGenerics'
+  excludeTest 'dagger.functional.GenericTest', 'packagePrivateTypeParameterDependencies'
+  excludeTest 'dagger.functional.GenericTest', 'publicSubclassWithPackagePrivateTypeParameterOfSuperclass'
+  excludeTest 'dagger.functional.GenericTest', 'singletonScopesAppliesToEachResolvedType'
+  excludeTest 'dagger.functional.GenericTest', 'testGenericComponentCreate'
+  excludeTest 'dagger.functional.GenericTest', 'testGenericDoubleReferences'
+  excludeTest 'dagger.functional.GenericTest', 'testGenericSimpleReferences'
```

Comment:
I'd like to try and see how complicated it is to reduce this list before merging. I fear the current design really only addresses one specific case.

Reply:
I was able to make the solution more generic and have two more tests pass. The rest of the test deal specifically with generic parameters being a part of a Module's inherited declaration! :( (See `ParentModule`)